doc/reference/trd104-syscalls.md

```diff
@@ -446,12 +532,12 @@ kernel.
 The register arguments for Subscribe system calls are as follows. The
 registers r0-r3 correspond to r0-r3 on CortexM and a0-a3 on RISC-V.
 
-| Argument            | Register |
-|---------------------|----------|
-| Driver number       | r0       |
-| Subscribe number    | r1       |
-| Upcall pointer      | r2       |
-| Application data    | r3       |
+| Argument            | Register | Type               |
+|---------------------|----------|--------------------|
+| Driver number       | r0       | `u32`              |
+| Subscribe number    | r1       | `u32`              |
+| Upcall pointer      | r2       | C_FUNCTION_POINTER |
```

Comment:
Is there a meaningful way to specify the function signature of this upcall pointer? Do we even document that today?

I'm asking because working on the isolated nonvolatile storage API, in light of this discussion, I am removing the length fields and using the allow sizes instead. However, it is still useful to be able to do a "get size" command that provides the number of bytes via an upcall. I want to future proof that as a `uint64_t`.

Reply:
The closest we come is this: https://github.com/tock/tock/blob/master/doc/reference/trd104-syscalls.md#52-subscribe

Which would put `void (subscribe_upcall)(int, int, int, void*);` in place there.

I think something more in the spirit of the rest of the documentation would be `fn(u32, u32, u32, POINTER_OR_ZERO) -> ()`. 

Or the exhaustive set of patterns to cover your use case might be: `fn({u32,i32,u64_lo,i64_lo}, {u32,i32,u64_hi,i64_hi}, {u32,i32}, POINTER_OR_ZERO) -> ()`?